// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.rpiledcontroller";
option java_outer_classname = "RPiLEDController";
option objc_class_prefix = "PIL";

package led_ctlr;

// The RPi controller service definition.
service RPi_Ctlr {
  // Sends a LED control message
  rpc SendMsg (RPiRequest) returns (RPiReply) {}
}

// The request message containing the user's name.
message RPiRequest {
  string name = 1;
  int32 colour = 2;
  int32 operation = 3;
}

// The response message containing the results
message RPiReply {
  string message = 1;
}
